modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2379,14 +2379,20 @@ def explode(self, column):
     def apply(self, func, axis, *args, **kwargs):
         # if any of args contain modin object, we should
         # convert it to pandas
+
+        # TODO(vkarthik): currently is_series only works for callable
+        # functions that return Series objects. We should handle this
```

Comment:
I'm not sure I follow? In the `_callable_func` docstring, `is_series` is documented as a flag that determines whether or not we're working with a Series, but here it seems to indicate that its a flag for when the function will return a Series? Perhaps we should update the docstring and this comment to something like `is_series` is a flag that indicates we are applying a function to a Series object that will result in a Dataframe?

Reply:
I remember why I called it `is_series` now. Technically the code I've added here should work for any case where we know we are trying to apply a function to a Series object. However, we should probably just use this code path for when we know what we are returning a Series object in the `func` being applied to an existing Series object. This will have to be documented clearly I think. There seems to be no way around this but we have to handle this weirdness from pandas.